### PR TITLE
Test PR

### PR DESCRIPTION
**Pull request by joachimprinzbach**
pong!

### CODE REVIEW

**Comment by joachimprinzbach** [comment]
test

**Comment by joachimprinzbach** [comment]
a

**Comment by joachimprinzbach** [comment]
hihi

**Comment by incubator-tekton** [comment]
generated

**Comment by incubator-tekton** [comment]
generated

**Comment by incubator-tekton** [comment]
generated

**Comment by incubator-tekton** [comment]
blabb

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(GitHubWebhookRestController.class);
+    private static final String GITHUB_EVENT_HEADER_KEY = "X-GitHub-Event";
+
+    @Autowired
+    private PullRequestCommentService commentService;
+
+    @PostMapping(path = "webhook/github")
+    public void handleGitHubWebhookEvent(@RequestBody String data, @RequestHeader(GITHUB_EVENT_HEADER_KEY) String githubEvent) {
+        Optional<GitHubEventType> gitHubEventType = GitHubEventType.fromEventName(githubEvent);
+        if (gitHubEventType.isPresent()) {
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java:42]: Hast du schon mindestens Java 9 Level? Dann kannst du nämlich [ifPresentOrElse ](http://iteratrlearning.com/java9/2016/09/05/java9-optional.html)verwenden.

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            }
+        } else {
+            LOGGER.info("Ignored GitHub Event Type: {}", githubEvent);
+        }
+    }
+
+    private void handlePullRequestEvent(JsonObject jsonObject) {
+        String action = getActionElementValue(jsonObject);
+        Optional<GitHubPullRequestEventAction> gitHubPullRequestEventAction = GitHubPullRequestEventAction.fromActionName(action);
+        if (gitHubPullRequestEventAction.isPresent()) {
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: Evtl. auslagern des if blocks in eine Methode und dann ifPresent(x-> callMethod(x)) verwenden.

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    private void handleIssueCommentEvent(JsonObject jsonObject) {
+        String action = getActionElementValue(jsonObject);
+        Optional<GitHubIssueCommentEventAction> gitHubIssueCommentEventAction = GitHubIssueCommentEventAction.fromActionName(action);
+        if (gitHubIssueCommentEventAction.isPresent()) {
+            switch (gitHubIssueCommentEventAction.get()) {
+                case CREATED: {
+                    System.out.println("Issue comment created");
+                    String asString = jsonObject.get("comment").getAsJsonObject().get("body").getAsString();
+                    if (asString.startsWith("/ping")) {
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: Geniales Feature :+1:

Comment (joachimprinzbach) [reply src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: :-D

**Comment by incubator-tekton** [comment]
blabb

**Comment by incubator-tekton** [comment]
blabb

**Comment by incubator-tekton** [comment]
blabb

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
Successfully deployed version latest

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
blabb

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
Successfully deployed version mywesometag

**Comment by incubator-tekton** [comment]
Successfully deployed version latest

**Comment by incubator-tekton** [comment]
blabb

**Comment by incubator-tekton** [comment]
blabb

**Comment by incubator-tekton** [comment]
2019-11-30T16:29:52.706

**Comment by incubator-tekton** [comment]
pong!

**Comment by joachimprinzbach** [comment]
/deploy feat-refactor_domain_model-c4604b75e7515feb1d15ff0cdacdfb8f4b90c22c-50.1

**Comment by incubator-tekton** [comment]
pong!

**Comment by joachimprinzbach** [comment]
/deploy feat-refactor_domain_model-d78668006c3b1cf19fb6a25de77c38faec1525ec-51.1

**Comment by incubator-tekton** [comment]
Successfully deployed version feat-refactor_domain_model-d78668006c3b1cf19fb6a25de77c38faec1525ec-51.1

**Comment by incubator-tekton** [comment]
pong!

**Comment by joachimprinzbach** [comment]
/deploy feat-refactor_domain_model-69ad3c22a0a94c6081e95feee37b3dd3040f5963-54.1

**Comment by incubator-tekton** [comment]
Successfully deployed version feat-refactor_domain_model-69ad3c22a0a94c6081e95feee37b3dd3040f5963-54.1

**Pull request merged**